src/server/frontend_xwayland/xwayland_wm_surface.cpp

```diff
@@ -266,17 +282,23 @@ void mf::XWaylandWMSurface::read_properties()
 
 void mf::XWaylandWMSurface::move_resize(uint32_t detail)
 {
-    if (detail == _NET_WM_MOVERESIZE_MOVE)
-        shell_surface->initiate_interactive_move();
-    else if (auto const edge = wm_resize_edge_to_mir_resize_edge(detail))
-        shell_surface->initiate_interactive_resize(edge.value());
-    else
-        mir::log_warning("XWaylandWMSurface::move_resize() called with unknown detail %d", detail);
+    aquire_shell_surface([detail](auto shell_surface)
+        {
+            if (detail == _NET_WM_MOVERESIZE_MOVE)
+                shell_surface->initiate_interactive_move();
+            else if (auto const edge = wm_resize_edge_to_mir_resize_edge(detail))
+                shell_surface->initiate_interactive_resize(edge.value());
+            else
+                mir::log_warning("XWaylandWMSurface::move_resize() called with unknown detail %d", detail);
+        });
```

Comment:
Should only the `shell_surface->...` calls be in the `aquire_shell_surface(` functor?

Reply:
Yes, done